src/io/GeantGeometryMap.cc

```diff
@@ -0,0 +1,109 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file GeantGeometryMap.cc
+//---------------------------------------------------------------------------//
+#include "GeantGeometryMap.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct/destruct with defaults
+ */
+GeantGeometryMap::GeantGeometryMap()  = default;
+GeantGeometryMap::~GeantGeometryMap() = default;
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return the mat_id for a given vol_id
+ */
+GeantGeometryMap::mat_id GeantGeometryMap::get_matid(vol_id& volume_id)
+{
+    auto iter = volid_to_matid_.find(volume_id);
+    REQUIRE(iter != volid_to_matid_.end());
+    return iter->second;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return the GeantVolume associated with the vol_id
+ */
+GeantVolume GeantGeometryMap::get_volume(vol_id& volume_id)
+{
+    auto iter = volid_to_volume_.find(volume_id);
+    REQUIRE(iter != volid_to_volume_.end());
+    return iter->second;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return the GeantMaterial associated with the mat_id
+ */
+GeantMaterial GeantGeometryMap::get_material(mat_id& material_id)
+{
+    auto iter = matid_to_material_.find(material_id);
+    REQUIRE(iter != matid_to_material_.end());
+    return iter->second;
+}
+//---------------------------------------------------------------------------//
+/*!
+ * Return the GeantElement associated with the elem_id
+ */
+GeantElement GeantGeometryMap::get_element(elem_id& element_id)
+{
+    auto iter = elemid_to_element_.find(element_id);
+    REQUIRE(iter != elemid_to_element_.end());
+    return iter->second;
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Return a copy of private member volid_to_matid_
+ */
+std::map<GeantGeometryMap::vol_id, GeantGeometryMap::mat_id>
+GeantGeometryMap::volid_to_matid_map()
+{
+    return volid_to_matid_;
```

Comment:
Return a const reference -- here you're making a very expensive copy.

Reply:
Agree with returning a reference instead of doing it by copy. But most importantly... do we need it? I only added the function for a quick test, that is still in the `geant-exporter-cat`, and that maybe we would want a direct access to this map in `Celeritas` (which could be just set public as well).

Reply:
I think it makes sense to have direct read-only access to it for output purposes. This class is just a piece of helper glue code as opposed to a low-level critical design, so we don't have to worry too much about it. It's kind of doing two things at once: being a builder class (constructing the maps) and a storage/access class (owning the maps). There's no danger to making the constructed maps fully readable, anyway.

Reply:
Sounds good. Done.